### ISSUE TITLE
Attempt to invoke virtual method 'int com.horcrux.svg.SvgViewShadowNode.hitTest(android.graphics.Point)' on a null object reference

### ISSUE THREAD

**Issue by 999946** [react-native-community/react-native-svg#695]
react-native : 0.53.0
manufacturer:HUAWEI
model:HUAWEI NXT-AL10
osName：android
osVersion：7.0

```
java.lang.NullPointerException Attempt to invoke virtual method 'int com.horcrux.svg.SvgViewShadowNode.hitTest(android.graphics.Point)' on a null object reference 
    SvgView.java:97 com.horcrux.svg.SvgView.dispatchTouchEvent
    ViewGroup.java:2671 android.view.ViewGroup.dispatchTransformedTouchEvent
    ViewGroup.java:2301 android.view.ViewGroup.dispatchTouchEvent
    ViewGroup.java:2671 android.view.ViewGroup.dispatchTransformedTouchEvent
    ViewGroup.java:2301 android.view.ViewGroup.dispatchTouchEvent
    ViewGroup.java:2671 android.view.ViewGroup.dispatchTransformedTouchEvent
    ViewGroup.java:2301 android.view.ViewGroup.dispatchTouchEvent
    ViewGroup.java:2671 android.view.ViewGroup.dispatchTransformedTouchEvent
    ViewGroup.java:2301 android.view.ViewGroup.dispatchTouchEvent
    ViewGroup.java:2671 android.view.ViewGroup.dispatchTransformedTouchEvent
    ViewGroup.java:2301 android.view.ViewGroup.dispatchTouchEvent
    ViewGroup.java:2671 android.view.ViewGroup.dispatchTransformedTouchEvent
    ViewGroup.java:2301 android.view.ViewGroup.dispatchTouchEvent
    ViewGroup.java:2671 android.view.ViewGroup.dispatchTransformedTouchEvent
    ViewGroup.java:2301 android.view.ViewGroup.dispatchTouchEvent
    ViewGroup.java:2671 android.view.ViewGroup.dispatchTransformedTouchEvent
    ViewGroup.java:2301 android.view.ViewGroup.dispatchTouchEvent
    DecorView.java:447 com.android.internal.policy.DecorView.superDispatchTouchEvent
    PhoneWindow.java:1871 com.android.internal.policy.PhoneWindow.superDispatchTouchEvent
    Dialog.java:949 android.app.Dialog.dispatchTouchEvent
    DecorView.java:409 com.android.internal.policy.DecorView.dispatchTouchEvent
    View.java:10232 android.view.View.dispatchPointerEvent
    ViewRootImpl.java:4862 android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent
    ViewRootImpl.java:4723 android.view.ViewRootImpl$ViewPostImeInputStage.onProcess
    ViewRootImpl.java:4255 android.view.ViewRootImpl$InputStage.deliver
    ViewRootImpl.java:4308 android.view.ViewRootImpl$InputStage.onDeliverToNext
    ViewRootImpl.java:4274 android.view.ViewRootImpl$InputStage.forward
    ViewRootImpl.java:4401 android.view.ViewRootImpl$AsyncInputStage.forward
    ViewRootImpl.java:4282 android.view.ViewRootImpl$InputStage.apply
    ViewRootImpl.java:4458 android.view.ViewRootImpl$AsyncInputStage.apply
    ViewRootImpl.java:4255 android.view.ViewRootImpl$InputStage.deliver
    ViewRootImpl.java:4308 android.view.ViewRootImpl$InputStage.onDeliverToNext
    ViewRootImpl.java:4274 android.view.ViewRootImpl$InputStage.forward
    ViewRootImpl.java:4282 android.view.ViewRootImpl$InputStage.apply
    ViewRootImpl.java:4255 android.view.ViewRootImpl$InputStage.deliver
    ViewRootImpl.java:6682 android.view.ViewRootImpl.deliverInputEvent
    ViewRootImpl.java:6656 android.view.ViewRootImpl.doProcessInputEvents
    ViewRootImpl.java:6617 android.view.ViewRootImpl.enqueueInputEvent
    ViewRootImpl.java:6809 android.view.ViewRootImpl$WindowInputEventReceiver.onInputEvent
    InputEventReceiver.java:192 android.view.InputEventReceiver.dispatchInputEvent
    MessageQueue.java:-2 android.os.MessageQueue.nativePollOnce
    MessageQueue.java:356 android.os.MessageQueue.next
    Looper.java:138 android.os.Looper.loop
    ActivityThread.java:6523 android.app.ActivityThread.main
    Method.java:-2 java.lang.reflect.Method.invoke
    ZygoteInit.java:942 com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run
    ZygoteInit.java:832 com.android.internal.os.ZygoteInit.main
```

**Comment by Barbiero** [comment]
I want to add that this also happens to me, with RN 0.56, Android 8.0, and RN-SVG 6.5.0. works fine on 6.4.1.

**Comment by jamesaarondevlin** [comment]
This is also happening for me. What side effects would there be in adding a null reference check to the result of getShadowNode()? If that's null can you return super.dispatchTouchEvent(ev)? Kinda like this? Apologies for maybe not seeing bigger picture or knowing what side effects this might produce but hoping to understand this a bit more.

```
@Override
    public boolean dispatchTouchEvent(MotionEvent ev) {
        SvgViewShadowNode  shadowNode = getShadowNode();
        if(shadowNode != null){
            mTargetTag = getShadowNode().hitTest(new Point((int) ev.getX(), (int) ev.getY()));

            if (mTargetTag != -1) {
                handleTouchEvent(ev);
                return true;
            }
        }

        return super.dispatchTouchEvent(ev);
    }
```

**Comment by msand** [comment]
Returning false should probably be fine, but would be more interesting to find out why there's no shadow node?

**Comment by adammcarth** [comment]
I'm experiencing this crash too on React `16.3.1` and React Native `0.55.4`.

**Comment by mtettmar** [comment]
I'm experiencing the same problem. I have an Svg inside a TouchableOpacity. First time I tap it, it works fine. If I then go to a new screen, then come back and tap it again, the app just closes. The log shows:

java.lang.NullPointerException: Attempt to invoke virtual method 'int com.horcrux.svg.SvgViewShadowNode.hitTest(android.graphics.Point)' on a null object reference

**Comment by anuraginwadi** [comment]
+1

**Comment by pibo** [comment]
+1

**Comment by MrLoh** [comment]
I also have this problem. Any release plan for the fix. This is pretty bad.

**Comment by msand** [comment]
Has anyone tried if the hit testing works correctly with the pr? Even after coming to a screen that is..

**Comment by MrLoh** [comment]
@msand I don’t understand your question hit testing?, that is..?), but the PR seems to fix this issue for us.

**Comment by msand** [comment]
Hit testing is process of walking the tree of nodes, to find the first (topmost) one which intersects the pointer when pressing the screen, i.e. the thing which is broken here. So, if you put a clickhandler on a svg element, does it still respond to touches correctly if you exit and reenter the view? @mtettmar can you test with the newly published 6.5.2?

**Comment by adammcarth** [comment]
@msand I can at least confirm that updating from `6.5.0` to `6.5.2` has fixed the crash I was experiencing when tapping on SVG elements. I wasn't specifying my own onPress handler, but I believe `react-native-shadow` may have been - which uses `react-native-svg` as a core dependancy. Tapping a drop shadow after switching screens was what caused the app to crash.
